src/app/util/chip-message-send.cpp

```diff
@@ -75,7 +79,19 @@ EmberStatus chipSendUnicast(NodeId destination, EmberApsFrame * apsFrame, uint16
     buffer->SetDataLength(dataLength);
 
     // TODO: temprary create a handle from node id, will be fix in PR 3602
-    CHIP_ERROR err = SessionManager().SendMessage({ destination, Transport::kAnyKeyId, 0 }, std::move(buffer));
+    Messaging::ExchangeContext * exchange = ExchangeManager().NewContext({ destination, Transport::kAnyKeyId, 0 }, nullptr);
+    if (exchange == nullptr)
+    {
+        return EMBER_DELIVERY_FAILED;
+    }
+
+    Messaging::SendFlags sendFlags;
+
+    sendFlags.Set(Messaging::SendMessageFlags::kFromInitiator, true).Set(Messaging::SendMessageFlags::kNoAutoRequestAck, true);
```

Comment:
Why do we need to set this initiator flag, as an API consumer?  Shouldn't the ExchangeContext know whether we're the initiator or not?  It sure looks like it does, and it ignores the "kFromInitiator" flag, as it should.  In fact, absolutely nothing uses that flag.   I filed https://github.com/project-chip/connectedhomeip/issues/5651 to do that.

Why do we want to force no ack here?  I don't see a reason for it....

Reply:
Here we defined a temporary Protocol TempZCL to carry over legacy ZCL messages, but unlike other standard protocols like Interaction Model, Echo ..., there is no protocol instances to host the conversation with exchange, such as EchoClient and EchoServer. 

Therefore, we used ephemeral exchange to send and receive ZCL messages and TempZCL Protocol does not know the semantics of each message. This is similar to the current implementation, just use an ephemeral exchange to send message and use its unsolicited message handler to receive messages.

We need to set flag kFromInitiator to allow receiver to deliver this message to corresponding unsolicited message handlers, and we also need to set flag kNoAutoRequestAck since there is no dedicated conversation to host the ack message.

Reply:
Hmm...  I can see this being ok very short-term, but it seems to me we should actually have a protocol instance so we can actually send the response back on the right exchange.  I'm OK with that happening in a followup, but:

1. There should be clear comments in the code explaining why we are doing these weird things, more or less summarizing the explanation here.
2. Those comments should reference an issue that we resolve in the near future which will remove he need for doing those weird things.

Reply:
Added. We already have the protocol instance implemented in IM protocol for this purpose. After we convert all legacy ZCL Ember messages to IM constructs, this logic will be cleaned up.